dataservice/api/common/validation.py

```diff
@@ -97,3 +97,39 @@ def validate_list(input_items):
                 f'valid set: {", ".join(valid_items)}'
             )
     return validate_list
+
+
+def alphanumeric_str_validation_generator(field_name, quantifier_regex='+'):
```

Comment:
Why do you need to take `quantifier_regex` as a parameter if it seems to always be constant?

Reply:
What do you mean by `always be constant`? If we have another field which requires alphanumeric str validation but has a different min/max length requirement, then you'd need to specify this to the validation func generator. Or am I not understanding the question?

Reply:
Do we have another field that requires it? If not, why not simply validate the length requirements statically instead of having an unecessary abstraction?

Reply:
We don't have one now, but I thought it doesn't hurt to implement this in a generic way so that it could be used in the future if needed

If we're going to make the length validation constant then we may as well get rid of the validation func generator and just implement a specific `validate_study_code` method in `dataservice/study/schema.py`